### DISCUSSION TITLE
q-table row index needed for shift-select

### DISCUSSION THREAD

**Discussion by CrawfordW** [quasarframework/quasar#10930] [General - Components / Directives / etc]
I have several tables with many columns. To save space, I don't want to use checkboxes for selection, but allow selection by just clicking anywhere on a row .  To implement selecting multiple rows using shift-click, I need to know the index of each row after sorting so I can determine whether rows are adjacent.
Is there a way to access the row index in the sorted array, or do I need to build this myself in the customSort function?

**Comment by ontwikkelfabriek** [comment]
@CrawfordW 

>  I need to know the index of each row after sorting so I can determine whether rows are adjacent.

Getting the row index is a problem when a q-table is sorted?

**Reply by CrawfordW** [reply]
Well I figured out how to do it in my customSort function...
```
customSort (rows, sortBy, descending) {
  const date = [...rows]
    data.sort((a, b) => {
       ....
    })
    // Set the new row indices:
    let rowIndex = 0
    data.forEach(row => {
        row.rowIndex = rowIndex
        rowIndex += 1
    })
    return data
}
```
